pandas/tests/indexes/multi/test_indexing.py

```diff
@@ -162,6 +162,34 @@ def test_putmask_multiindex_other(self):
         expected = MultiIndex.from_tuples([right[0], right[1], left[2]])
         tm.assert_index_equal(result, expected)
 
+    def test_putmask_keep_dtype(self, any_numeric_ea_dtype):
```

Comment:
Are the two test cases also covering for the all-nan case in https://github.com/pandas-dev/pandas/issues/49830#issue-1459159450 or do you think it's not necessary?

Reply:
The variable might be a misnomer then `any_numeric_ea_dtype` (if it covers all nans, this would also include non EA dtype).

Reply:
It covers the case in a sense that the current tests represent the all nan case as well, I’ll add a specific test when actually fixing the issue anyway